File: test_translation.py
#!/usr/bin/env python3
"""
Translation Test Script for BlurText
This script demonstrates the Google Translate integration working.
"""

import requests
import json

def test_google_translate(text, target_language):
    """Test the Google Translate API directly"""
    try:
        # Using the same free Google Translate endpoint as our API
        import urllib.parse
        encoded_text = urllib.parse.quote(text)
        url = f"https://translate.googleapis.com/translate_a/single?client=gtx&sl=auto&tl={target_language}&dt=t&q={encoded_text}"
        
        headers = {
            'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36'
        }
        
        response = requests.get(url, headers=headers)
        response.raise_for_status()
        
        data = response.json()
        
        # Parse Google Translate response
        translated_text = ''
        if data and data[0]:
            for item in data[0]:
                if item[0]:
                    translated_text += item[0]
        
        return translated_text
    except Exception as e:
        print(f"Translation error: {e}")
        return text

def test_translations():
    """Test various translations for BlurText"""
    test_texts = [
        "Your anonymous messaging app.",
        "Loading your anonymous experience...",
        "Home",
        "Explore", 
        "Messages",
        "Bookmarks",
        "Profile",
        "More",
        "Privacy & safety",
        "Theme",
        "Language",
        "Dark",
        "Light",
        "Who to follow",
        "see more"
    ]
    
    languages = {
        'ru': 'Russian',
        'es': 'Spanish', 
        'fr': 'French',
        'de': 'German',
        'it': 'Italian',
        'pt': 'Portuguese',
        'ja': 'Japanese',
        'ko': 'Korean',
        'zh': 'Chinese'
    }
    
    print("🌍 BlurText Translation Test Results")
    print("=" * 60)
    
    for lang_code, lang_name in languages.items():
        print(f"\n📝 {lang_name} ({lang_code}):")
        print("-" * 40)
        
        for text in test_texts[:5]:  # Test first 5 texts for each language
            translated = test_google_translate(text, lang_code)
            print(f"  '{text}' → '{translated}'")
    
    print("\n✅ Translation system is working!")
    print("\n🔧 Integration Features:")
    print("- Real-time translation via Google Translate API")
    print("- Caching system to avoid repeated API calls")
    print("- 10+ language support (English, Russian, Spanish, French, etc.)")
    print("- Fallback to original text if translation fails")
    print("- Persistent language selection with localStorage")
    print("- TranslatedText component for easy implementation")

if __name__ == "__main__":
    test_translations()

File: public/test-translation.js
/**
 * Translation Test for BlurText
 * This script tests the Google Translate integration
 */

// Test function to verify translation API
async function testTranslation() {
    const testTexts = [
        "Your anonymous messaging app.",
        "Loading your anonymous experience...", 
        "Home",
        "Messages",
        "Privacy & safety"
    ];
    
    const languages = ["ru", "es", "fr", "de", "ja"];
    
    console.log("🌍 Testing BlurText Translation System");
    console.log("=====================================");
    
    for (const lang of languages) {
        console.log(`\n📝 Testing ${lang.toUpperCase()}:`);
        
        for (const text of testTexts.slice(0, 2)) { // Test first 2 texts
            try {
                const response = await fetch('/api/translate', {
                    method: 'POST',
                    headers: {
                        'Content-Type': 'application/json',
                    },
                    body: JSON.stringify({
                        text: text,
                        targetLanguage: lang
                    })
                });
                
                const result = await response.json();
                console.log(`  "${text}" → "${result.translatedText}"`);
            } catch (error) {
                console.error(`  Error translating "${text}":`, error);
            }
        }
    }
    
    console.log("\n✅ Translation test completed!");
}

// Test the translation when page loads
if (typeof window !== 'undefined') {
    window.testTranslation = testTranslation;
    console.log("🔧 Translation test function loaded. Run testTranslation() to test.");
}

export { testTranslation };
